setup.py

```diff
@@ -33,12 +33,13 @@
     author="Apache",
     author_email="dev@skywalking.apache.org",
     license="Apache 2.0",
-    packages=find_packages(exclude=("tests",)),
+    packages=find_packages(exclude=("tests", "tests.*")),
     include_package_data=True,
     install_requires=[
         "grpcio",
         "grpcio-tools",
         "packaging",
+        "requests",
```

Comment:
Hi @tom-pytel I missed this in this PR, but this makes `requests` a mandatory dependency of skywalking-python, please also take a look at https://github.com/apache/skywalking/issues/7282 that `requests` depends on a LGPL licensed dependency that we cannot ship with in ASF project. As we have this in `extras_require/http`, can we just remove this? When users want to use `http` protocol, they can use something like `pip install skywalking-python[http]`.

FYI @wu-sheng

Reply:
Why a plugin requires an agent-level dependency?

Reply:
> Why a plugin requires an agent-level dependency?

We support grpc and http protocols, for http protocol, we use `requests` to send http requests, as we use grpc as default protocol and http is optional (can be installed by `pip install skywalking-python[http]`), I think @tom-pytel missed that and wanted to test http protocol so he add the dependency here

Reply:
OK, get it. It is glad we don't really depend on it.

Reply:
Sure if it is problematic then we remove it from the required dependencies if the license will cause problems. I could also look into using a different communication method like `urllib.request` or `urllib3.request`?

As for http protocol, we are doing stress testing here and finding that grpc is not entirely reliable and the http protocol is actually a lot more stable. Not sure why this is happening, maybe grpc is not configured correctly or the timeouts are causing problems. But the main result is that you should consider the http protocol a little more than just optional at this point since it is capable of working in scenarios for us where grpc breaks.

Reply:
@tom-pytel Could you share how you test the performance in another separate issue? From the last several weeks' perf tests, the JSON really doesn't have good performance from a Java perspective, tested in the OAP backend.